include/circt/Dialect/FIRRTL/FIRRTLTypesImpl.td

```diff
@@ -143,17 +143,30 @@ def FVectorTypeImpl : FIRRTLImplType<"FVector", [FieldIDTypeInterface]> {
 def BundleImpl : FIRRTLImplType<"Bundle", [FieldIDTypeInterface]> {
   let summary = "an aggregate of named elements. This is effectively a struct.";
   let genStorageClass = false;
-  let parameters = (ins "ArrayRef<BundleElement>":$elements);
+  let parameters = (ins "ArrayRef<BundleElement>":$elements, "StringAttr":$bundleName);
```

Comment:
Changing Bundles to have names and descriptions seems like something to be introduced separately, and with tests / examples?  Exciting, something we've talked about for a while.  Are names of bundle types unique? Are descriptions limited in some way, should they be?

In talks about supporting this sort of thing, there was hope to avoid copying the full type + description around-- not that this has to be addressed here, but I think is something we want to be intentional about as I'm pretty sure we'd like to support named bundles of some sort "soon".
cc @youngar .

Can these new bundles be used anywhere else, or only interfaces?  Should this be a separate type for use in interface "bodies"?

What is valid to go inside an interface body? If it's not everything suitable for a module, please add a verifier/constraints/something.